src/backend/catalog/dependency.c

```diff
@@ -1359,6 +1359,21 @@ AcquireDeletionLock(const ObjectAddress *object, int flags)
 			LockRelationOid(object->objectId, ShareUpdateExclusiveLock);
 		else
 			LockRelationOid(object->objectId, AccessExclusiveLock);
+
+		/*
+		 * GPDB: If this was a partition, or some other object for which
+		 * we are careful to always lock the parent object, we don't need
+		 * to keep the lock on the sub-object. This helps to keep the lock
+		 * table size in check, if you e.g. drop a table with thousands
+		 * of partitions. It would be unpleasent if you could not drop
+		 * such a table because the lock table runs out of space.
+		 *
+		 * To provide at least some protection though, we still actuire
+		 * the lock momentarily.
+		 */
+		if (!gp_keep_partition_children_locks &&
```

Comment:
Let's preserve the DDL's patch behaviour. Just provide correct backport of upstream patch for ORCA.